### PR TITLE
Fix remaining flakey tests and add cluster domain test

### PR DESCRIPTION
**Pull request by rancher-max**
- Addresses all flakey tests from and resolve https://github.com/rancher/rio/issues/749
- Adds a validation test for custom cluster domain, which asserts the following:
  - http and https endpoints are available on all services, whether created before or after the clusterdomain was.
  - http and https custom domain endpoints retrieve the expected response
- Minor updates to how it's running the tests in docker so that Jenkins reports pass/fail correctly

### CODE REVIEW

**Comment by rancher-max** [comment]
I think this is in a good state, but I'm going to be continually re-running the build until I see a failure. If the failure is due to Drone issues (spot instances for example) then I'm ignoring it. Otherwise I'll fix the issue.

Feel free to rerun any builds of this PR you see that have passed!

File: tests/validation/domain_test.go [hunk 1]
```diff
@@ -14,33 +15,85 @@ func domainTests(t *testing.T, when spec.G, it spec.S) {
...
+			}
+			domain.ApplyClusterDomain(ip)
+			otherService.Create(t, "ibuildthecloud/demo:v3")
+		})
+
+		it.After(func() {
+			otherService.Remove()
+			if os.Getenv("RIO_A_RECORD") != "" {
+				testutil.DeleteRoute53DNS(ip)
+			}
```
Comment (cbron) [inline tests/validation/domain_test.go]: Do you need to clean up cert ?

File: tests/testutil/publicdomain.go [hunk 2]
```diff
@@ -82,6 +83,35 @@ func (td *TestDomain) GetKubeDomain() string {
 	return getStandardFormatDomain(results)
 }
 
+//ApplyClusterDomain creates and applies a custom clusterdomain
+func (td *TestDomain) ApplyClusterDomain(ip string) {
+	valToApply := []byte(`apiVersion: admin.rio.cattle.io/v1
+kind: ClusterDomain
+metadata:
+  name: ` + GetAInfo() + `
```
[the remark this answers is not in the document]
Comment (rancher-max) [reply tests/testutil/publicdomain.go]: It shouldn't ever be an empty string I think.. if it is that's an error in how the test is called

File: tests/testutil/testutil.go [hunk 3]
```diff
@@ -266,56 +287,110 @@ func homeDir() string {
 	return os.Getenv("USERPROFILE") // windows
 }
 
-// CreateCNAME creates a CNAME if it doesn't exist or update if it exist already in the DNS Zone
+//CreateSelfSignedCert creates public and private keys for a self signed certificate
+func CreateSelfSignedCert() {
+	key, err := rsa.GenerateKey(cryptRand.Reader, 2048)
+	if err != nil {
+		fmt.Printf("Private key cannot be created. %v\n", err.Error())
+		os.Exit(1)
```
[the remark this answers is not in the document]
Comment (rancher-max) [reply tests/testutil/testutil.go]: I can change that. Was trying to keep consistent with the rest of the file which is primarily doing os.Exit

File: tests/testutil/testutil.go [hunk 4]
```diff
@@ -266,56 +287,110 @@ func homeDir() string {
 	return os.Getenv("USERPROFILE") // windows
 }
 
-// CreateCNAME creates a CNAME if it doesn't exist or update if it exist already in the DNS Zone
+//CreateSelfSignedCert creates public and private keys for a self signed certificate
+func CreateSelfSignedCert() {
```
[the remark this answers is not in the document]
Comment (rancher-max) [reply tests/testutil/testutil.go]: Yep, for custom cluster domains to retrieve an https endpoint on services

File: tests/validation/domain_test.go (hunk 1 above)
Comment (rancher-max) [reply tests/validation/domain_test.go]: Ah yes I probably should. Added them to gitignore but will fix ipOrDns fmt issue and clean these up

File: tests/testutil/testutil.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (cbron) [reply tests/testutil/testutil.go]: I see it happens a lot in the file. t.fatal has some built in logging and will still print out results, not sure if os.exit does same but I would doubt it.

[the remark this answers is not in the document]
Comment (rancher-max) [reply tests/testutil/testutil.go]: I'll update all of them!

File: tests/testutil/service.go [hunk 5]
```diff
@@ -553,15 +553,16 @@ func (ts *TestService) GetKubeEndpointURLs() []string {
...
-func (ts *TestService) GetKubeFirstClusterDomain() string {
+func (ts *TestService) GetKubeFirstClusterDomain() (string, string) {
 	args := []string{"get", "clusterdomains",
-		"-o", `jsonpath="{.items[0].metadata.name}"`}
+		"-o", `jsonpath='{.items[0].metadata.name}{"\t"}{.items[0].spec.addresses[0].ip}'`}
 	clusterDomain, err := KubectlCmd(args)
 	if err != nil {
 		ts.T.Fatalf("Failed to get the first Cluster Domain:  %v", err.Error())
-		return ""
+		return "", ""
```
Comment (cbron) [inline tests/testutil/service.go]: Is this supposed to be double quoted ?

Comment (rancher-max) [reply tests/testutil/service.go]: I added a return value to the function, but actually I don't think that will ever get called due to the Fatal before it, so I can remove that return entirely

**Comment by rancher-max** [comment]
Latest change after approval just had an used function that I removed

**Pull request merged**